### PR TITLE
Allows suit storage's to put clothes on for you

### PR DESCRIPTION
**Pull request by Super3222**
:cl: Super3222
rscadd: Dragging yourself onto suit storage's will put on the equipment for you.
tweak: Emagging a suit storage will activate it's super UV. Letting suit storage's put on your equipment for you, will roast whoever's inside of it alive.
/:cl:

### CODE REVIEW

File: code/game/machinery/suit_storage_unit.dm [hunk 1]
```diff
@@ -326,6 +338,39 @@
 
 	return ..()
 
+/obj/machinery/suit_storage_unit/emag_act(mob/user)
+	if(locked) // under this case, it backfires.
+		if(!emagged)
+			emagged = TRUE
+		else
+			spawn(2)
```
Comment (Alek2ander) [inline code/game/machinery/suit_storage_unit.dm]: why?

File: code/game/machinery/suit_storage_unit.dm [hunk 2]
```diff
@@ -183,32 +181,146 @@
...
+	var/mob/living/carbon/human/H = user
+	var/turf/T = get_turf(src)
+
+	locked = TRUE
+
+	if(suit)
+		var/obj/outfit = null
+		if(H.wear_suit)
+			outfit = H.wear_suit
+			H.unEquip(outfit)
```
Comment (Alek2ander) [inline code/game/machinery/suit_storage_unit.dm:232]: What if it fails (suit is NODROP)?

File: code/game/machinery/suit_storage_unit.dm [hunk 3]
```diff
@@ -183,32 +181,146 @@
...
+
+		if(outfit)
+			T.contents += suit
+
+		H.equip_to_slot_or_del(suit, slot_wear_suit)
```
Comment (Alek2ander) [inline code/game/machinery/suit_storage_unit.dm]: Don't delete stuff for no reason

File: code/game/machinery/suit_storage_unit.dm [hunk 4]
```diff
@@ -183,32 +181,146 @@
...
+			if(H.shoes)
+				shoes = H.shoes
+				H.unEquip(shoes)
+
+			if(shoes)
+				T.contents += shoes
+
+			H.equip_to_slot_or_del(shoes, slot_shoes)
+			storage = shoes
+			playsound(src, 'sound/items/rped.ogg', 50, 1)
```
Comment (Alek2ander) [inline code/game/machinery/suit_storage_unit.dm]: What about gas tanks?

File: code/game/machinery/suit_storage_unit.dm [hunk 5]
```diff
@@ -183,32 +181,146 @@
...
+
+	locked = FALSE
+	open_machine(0)
+
+	H.loc = T
+	H.reset_perspective(null)
+	H.update_canmove()
+
+	spawn(10)
+		playsound(src.loc, 'sound/machines/ding.ogg', 50, 1)
```
Comment (Alek2ander) [inline code/game/machinery/suit_storage_unit.dm:332]: You're playing like 6 sounds at the same time. And then a ding a second AFTER the person is ejected.

File: code/game/machinery/suit_storage_unit.dm [hunk 6]
```diff
@@ -183,32 +181,146 @@
...
+
+/obj/machinery/suit_storage_unit/examine(mob/user)
+	..()
+	user << "<span class='blue'>A new system was installed into [src] to make putting on suits faster! (Simply drag yourself into the machine while it's open)."
```
Comment (Alek2ander) [inline code/game/machinery/suit_storage_unit.dm]: This should be in the changelog, not here

File: code/game/machinery/suit_storage_unit.dm [hunk 7]
```diff
@@ -183,32 +181,146 @@
...
+
+
+/obj/machinery/suit_storage_unit/emag_act(mob/user)
+	if(!emagged)
+		uv_super = TRUE
+		user << "<span class='warning'>You activate the UV's super setting.</span>"
+	else if (!uv_super)
+		uv_super = TRUE
```
Comment (Alek2ander) [inline code/game/machinery/suit_storage_unit.dm]: You don't actually set emagged

File: code/game/machinery/suit_storage_unit.dm [hunk 8]
```diff
@@ -183,32 +181,146 @@
...
-	if(occupant || helmet || suit || storage)
+	if(occupant)
 		user << "<span class='warning'>It's too cluttered inside to fit in!</span>"
 		return
 
+	var/enterdelay = 5
+
 	if(target == user)
-		user.visible_message("<span class='warning'>[user] starts squeezing into [src]!</span>", "<span class='notice'>You start working your way into [src]...</span>")
+		user.visible_message("<span class='warning'>[user] begins to enter into [src]!</span>", "<span class='notice'>You start positioning yourself inside [src]...</span>")
```
Comment (Alek2ander) [inline code/game/machinery/suit_storage_unit.dm]: It's half a second, this alert isn't needed

File: code/game/machinery/suit_storage_unit.dm (hunk 4 above)
Comment (Super3222) [inline code/game/machinery/suit_storage_unit.dm]: Fair point.

File: code/game/machinery/suit_storage_unit.dm (hunk 3 above)
Comment (Super3222) [inline code/game/machinery/suit_storage_unit.dm]: Right, I'll make a check for whether there's a suit or not.

File: code/game/machinery/suit_storage_unit.dm (hunk 5 above)
Comment (Super3222) [inline code/game/machinery/suit_storage_unit.dm:332]: F - U - N

File: code/game/machinery/suit_storage_unit.dm (hunk 3 above)
Comment (Alek2ander) [inline code/game/machinery/suit_storage_unit.dm]: There's a can_equip proc or something like that

**Pull request closed without merging**